colorbleed/plugins/maya/publish/validate_rendersettings.py

```diff
@@ -21,27 +22,31 @@ class ValidateResources(pyblish.api.InstancePlugin):
     hosts = ["maya"]
     families = ["colorbleed.renderlayer"]
 
+    DEFAULT_PADDING = 4
+    DEFAULT_PREFIX = {
+            "vray": "<Scene>/<Scene>_<Layer>/<Layer>",
+            "arnold": "<Scene>/<Scene>_<RenderLayer>/<RenderLayer>"
+    }
+
     def process(self, instance):
 
-        renderer = cmds.getAttr("defaultRenderGlobals.currentRenderer")
+        renderer = instance.data['renderer']
+        layer_node = instance.data['setMembers']
 
-        default_padding = 4
-        default_prefix = {
-            "vray": "<Scene>/<Scene>_<Layer>/<Layer>",
-            "arnold": "<Scene>/<Scene>_<RenderLayer>/<RenderLayer>"
-        }
-
-        if renderer == "vray":
-            prefix = cmds.getAttr("vraySettings.fileNamePrefix")
-            padding = cmds.getAttr("vraySettings.fileNamePadding")
-        else:
-            prefix = cmds.getAttr("defaultRenderGlobals.fileNamePrefix")
-            padding = cmds.getAttr("defaultRenderGlobals.fileNamePadding")
-
-        filename_prefix = default_prefix[renderer]
-        assert padding == default_padding, AttributeError(
-            "Expecting padding of 4 ( 0000 )"
-        )
-        assert prefix == filename_prefix, AttributeError(
+        # Collect the filename prefix in the renderlayer
+        with lib.renderlayer(layer_node):
+            if renderer == "vray":
+                prefix = cmds.getAttr("vraySettings.fileNamePrefix")
+                padding = cmds.getAttr("vraySettings.fileNamePadding")
+            else:
+                prefix = cmds.getAttr("defaultRenderGlobals.fileNamePrefix")
+                padding = cmds.getAttr("defaultRenderGlobals.fileNamePadding")
+
+        filename_prefix = self.DEFAULT_PREFIX[renderer]
```

Comment:
I think we should use `.get(renderer, "<Scene>/<Scene>_<RenderLayer>/<RenderLayer>")` to ensure it will pass when submitting with Maya Software for example.

Reply:
Yes, we should have a default. 

Actually, Arnold behaves just like the default renderers; only V-Ray seems to be behave differently. Maybe we should just default to `default` always, unless there's specific behavior to be implemented, like for V-Ray. That way the other behavior isn't linked to "arnold", but just used as default.